docs/core/tools/project-json.md

```diff
@@ -163,575 +82,977 @@ For example:
+Type: Boolean
+
+**true** to enable server garbage collection; otherwise, **false**. The default is **false**.
 
 For example:
 
-    "compilationOptions": {
-      "allowUnsafe": true
+    {
+        "runtimeOptions": {
+            "configProperties": {
+                "System.GC.Server": true
+            }
+        }
     }
 
-<a name="define"></a>
-### define
-Type: String[]
+#### System.GC.Concurrent
+Type: Boolean
 
-A list of defines such as "DEBUG" or "TRACE" that can be used in conditional compilation in the code.
+**true** to enable concurrent garbage collection; otherwise, **false**. The default is **false**.
 
 For example:
 
     {
-        "compilationOptions": {
-            "define": ["TEST", "OTHERCONDITION"]
+        "runtimeOptions": {
+            "configProperties": {
+                "System.GC.Concurrent": true
+            }
         }
     }
 
-<a name="nowarn"></a>
-### nowarn
-Type: String[]
+#### System.GC.RetainVM
+Type: Boolean
 
-A list of warnings to ignore.
+**true** to put segments that should be deleted on a standby list for future use instead of releasing them back to the operating system (OS); otherwise, **false**. The default is **false** (?).
 
 For example:
 
     {
-        "compilationOptions": {
-            "nowarn": ["CS0168", "CS0219"]
+        "runtimeOptions": {
+            "configProperties": {
+                "System.GC.RetainVM": true
+            }
         }
     }
 
-This ignores the warnings `The variable 'var' is assigned but its value is never used` and `The variable 'var' is assigned but its value is never used`
+#### System.Threading.ThreadPool.MinThreads
+Type: Integer
 
-<a name="additionalArguments"></a>
-### additionalArguments
-Type: String[]
+Overrides the number of minimum threads for the ThreadPool worker pool.
 
-A list of extra arguments that will be passed to the compiler.
+    {
+        "runtimeOptions": {
+            "configProperties": {
+                "System.Threading.ThreadPool.MinThreads": 4
+            }
+        }
+    }
 
-For example:
+#### System.Threading.ThreadPool.MaxThreads
+Type: Integer
+
+Overrides the number of maximum threads for the ThreadPool worker pool.
 
     {
-        "compilationOptions": {
-            "additionalArguments": ["/parallel", "/nostdlib"]
+        "runtimeOptions": {
+            "configProperties": {
+                "System.Threading.ThreadPool.MaxThreads": 25
+            }
         }
     }
 
-<a name="languageVersion"></a>
-### languageVersion
+### framework
+Type: Object
+
+Contains shared framework properties to use when activating the application. The presence of this section indicates that the application is a portable app designed to use a shared redistributable framework.
+
+#### name
 Type: String
 
-The version of the language used by the compiler: ISO-1, ISO-2, 3, 4, 5, 6, or Default
+Name of the shared framework.
 
-For example:
+    {
+        "runtimeOptions": {
+            "framework": {
```

Comment:
The `framework` is discussed in this link that existed as part of the announcements materials.
https://github.com/dotnet/cli/blob/v1.0.0-rc2/Documentation/specs/runtime-configuration-file.md#runtimeoptions-section-runtimeconfigjson

This is part where I found that link:
More configProperties can be found here, and a more detailed description of the runtimeOptions section as a whole can be found here.

On rel1.0.0 version, it also has a new field called applyPatches.
https://github.com/dotnet/cli/blob/rel/1.0.0/Documentation/specs/runtime-configuration-file.md

Who knows what's the right fields @blackdwarf? Should I add applyPatches too?